src/js/actions/libraries.js

```diff
@@ -335,17 +337,20 @@ define(function (require, exports) {
             return Promise.resolve();
         }
 
-        var docRef = docAdapter.referenceBy.id(currentDocument.id),
-            location = { x: 100, y: 100 },
-            representation = element.getPrimaryRepresentation();
-
-        return Promise.fromNode(function (cb) {
-            representation.getContentPath(cb);
-        }).then(function (path) {
-            var placeObj = libraryAdapter.placeElement(docRef, element, path, location);
-
-            return descriptor.playObject(placeObj);
-        });
+        return Promise
+                .fromNode(function (cb) {
+                    element.getPrimaryRepresentation().getContentPath(cb);
+                })
+                .bind(this)
+                .then(function (path) {
+                    var docRef = docAdapter.referenceBy.id(currentDocument.id),
+                        placeObj = libraryAdapter.placeElement(docRef, element, path, location);
+
+                    return descriptor.playObject(placeObj);
+                })
+                .then(function () {
+                    this.flux.actions.documents.updateDocument();
```

Comment:
Is there a way we can do this without an updateDocument call? Can we get the layer and pass this to addLayer somehow?